src/components/Topic/Record.tsx

```diff
@@ -1,23 +1,65 @@
-import React, { FunctionComponent } from "react";
-import { KafkaRecord } from "../../kafka/message-from-server";
+import React, { FunctionComponent, useCallback, useState } from "react";
+import { EnrichedKafkaRecord } from "../../overmind/connection/state";
+import classNames from "classnames";
 
 export interface RecordProps {
-  record: KafkaRecord;
+  record: EnrichedKafkaRecord;
 }
 
 export const Record: FunctionComponent<RecordProps> = ({ record }) => {
+  const [collapsed, setCollapsed] = useState(true);
```

Comment:
Instead of toggling collapse in-place, consider opening a modal with record contents to allow for better inspection of the key/value.